### PR TITLE
Fix deprecated takebuf_string method

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -111,12 +114,12 @@ end
...
     @test_throws ArgumentError ZlibInflateOutputStream(UInt8[], bufsize=0)
 
-    deflated = read(ZlibDeflateInputStream("foo".data))
+    deflated = read(ZlibDeflateInputStream(Vector{UInt8}("foo")))
     buf = IOBuffer()
     out = ZlibInflateOutputStream(buf)
     BufferedStreams.writebytes(out, deflated, length(deflated), true)
     flush(out)
-    @test takebuf_string(buf) == "foo"
+    @test String(take!((buf))) == "foo"
```
Comment (tkelman) [inline test/runtests.jl:122]: any time you fix a deprecation like this in a way that relies on Compat, you **must** verify that the minimum required Compat version implements the features you're using, and increment it if it doesn't

Comment (musm) [reply test/runtests.jl:122]: Right now it's Compat 0.8.0, not sure if that the minimum. I am strongly in favor of just using `Compat` in the REQUIRE file.

Comment (musm) [reply test/runtests.jl:122]: Even @ViralBShah has mentioned how big a pain it is to try to determine the minimal compat version......

Comment (tkelman) [reply test/runtests.jl:122]: Viral's wrong on this one. Not being correct about this means anyone who's pinning package versions, which we happen to do even on Compat in cases where reproducibility is important, will get broken packages any time you don't get the dependency requirements right. Check blame on the Compat readme, it's not hard.

**Pull request merged**